Servers/WindowServer/WSWindowManager.cpp

```diff
@@ -692,46 +705,53 @@ CElapsedTimer& WSWindowManager::DoubleClickInfo::click_clock(MouseButton button)
 
 // #define DOUBLECLICK_DEBUG
 
-void WSWindowManager::deliver_mouse_event(WSWindow& window, WSMouseEvent& event)
+void WSWindowManager::process_event_for_doubleclick(WSWindow& window, WSMouseEvent& event)
```

Comment:
This abstraction is needed because the drag MouseUp doesn't go through deliver_mouse_event, but process_ongoing_window_drag.

Reply:
Ah I see. Yeah, that's a bit messy. Maybe over time we could iterate on this so things flow a bit more clearly, but I think it's fine for now.